src/lib.rs

```diff
@@ -18,13 +18,13 @@ use std::net::{TcpListener, TcpStream};
 /// |-----------|--------------------------------------------|---------------------------------------------------------------------|
 /// | `handler` | `fn(Request<&[u8]>, &mut Response<&[u8]>)` | This function uses Types that are re-exported from the `http` crate |
 pub struct Server {
-    handler: fn(Request<&[u8]>, &mut Response<&[u8]>),
+    handler: fn(Request<&[u8]>, http::response::Builder) -> Response<&[u8]>,
```

Comment:
perhaps we can do `use http::response::Builder as ResponseBuilder;` here like in the example